homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -48,47 +66,115 @@ async def _async_update_data(self):
+            forecast_arg = "forecast_daily"
+        forecast_weather = [
+            self._convert_forecast(x) for x in getattr(weather_response, forecast_arg)
+        ]
+
         return {
-            ATTR_API_TEMPERATURE: weather_response.get_temperature("celsius").get(
-                "temp"
-            ),
-            ATTR_API_PRESSURE: weather_response.get_pressure().get("press"),
-            ATTR_API_HUMIDITY: weather_response.get_humidity(),
-            ATTR_API_WIND_BEARING: weather_response.get_wind().get("deg"),
-            ATTR_API_WIND_SPEED: weather_response.get_wind().get("speed"),
-            ATTR_API_CLOUDS: weather_response.get_clouds(),
-            ATTR_API_RAIN: self._get_rain(weather_response.get_rain()),
-            ATTR_API_SNOW: self._get_snow(weather_response.get_snow()),
-            ATTR_API_WEATHER: weather_response.get_detailed_status(),
-            ATTR_API_CONDITION: self._get_condition(
-                weather_response.get_weather_code()
+            ATTR_API_TEMPERATURE: current_weather.temperature("celsius").get("temp"),
+            ATTR_API_PRESSURE: current_weather.pressure.get("press"),
+            ATTR_API_HUMIDITY: current_weather.humidity,
+            ATTR_API_WIND_BEARING: current_weather.wind().get("deg"),
+            ATTR_API_WIND_SPEED: current_weather.wind().get("speed"),
+            ATTR_API_CLOUDS: current_weather.clouds,
+            ATTR_API_RAIN: self._get_rain(current_weather.rain),
+            ATTR_API_SNOW: self._get_snow(current_weather.snow),
+            ATTR_API_WEATHER: current_weather.detailed_status,
+            ATTR_API_CONDITION: self._get_condition(current_weather.weather_code),
+            ATTR_API_WEATHER_CODE: current_weather.weather_code,
+            ATTR_API_FORECAST: forecast_weather,
+        }
+
+    def _convert_forecast(self, entry):
+        forecast = {
+            ATTR_FORECAST_TIME: entry.reference_time("unix") * 1000,
+            ATTR_FORECAST_PRECIPITATION: self._calc_precipitation(
+                entry.rain, entry.snow
             ),
-            ATTR_API_WEATHER_CODE: weather_response.get_weather_code(),
+            ATTR_FORECAST_WIND_SPEED: entry.wind().get("speed"),
+            ATTR_FORECAST_WIND_BEARING: entry.wind().get("deg"),
+            ATTR_FORECAST_CONDITION: self._get_condition(entry.weather_code),
         }
 
+        temperature_dict = entry.temperature("celsius")
+        if "max" in temperature_dict and "min" in temperature_dict:
+            forecast[ATTR_FORECAST_TEMP] = entry.temperature("celsius").get("max")
+            forecast[ATTR_FORECAST_TEMP_LOW] = entry.temperature("celsius").get("min")
+        elif "temp" in temperature_dict:
+            forecast[ATTR_FORECAST_TEMP] = entry.temperature("celsius").get("temp")
+
+        return forecast
+
     @staticmethod
     def _get_rain(rain):
-        if "1h" in rain:
-            return round(rain["1h"], 0)
+        if "all" in rain:
```

Comment:
the key "1h" exists still hourly api. Therefore we should leave the if here and only add the new "all" if.
Have you tested, that your code is still working with the daily and hourly api?

Reply:
Ah, I thought that had gone away with the update to `pyowm==3.0.0` but you're right. Fixing it.